src/NHibernate/Linq/Functions/StringGenerator.cs

```diff
@@ -242,12 +242,17 @@ public class TrimGenerator : BaseHqlGeneratorForMethod
 		public TrimGenerator()
 		{
 			SupportedMethods = new[]
-									{
-										ReflectHelper.GetMethodDefinition<string>(s => s.Trim()),
-										ReflectHelper.GetMethodDefinition<string>(s => s.Trim('a')),
-										ReflectHelper.GetMethodDefinition<string>(s => s.TrimStart('a')),
-										ReflectHelper.GetMethodDefinition<string>(s => s.TrimEnd('a'))
-									};
+			{
+				ReflectHelper.GetMethodDefinition<string>(s => s.Trim()),
+				ReflectHelper.GetMethodDefinition<string>(s => s.Trim('a')),
+				ReflectHelper.GetMethodDefinition<string>(s => s.Trim('a', 'a')),
+				ReflectHelper.GetMethodDefinition<string>(s => s.TrimStart()),
+				ReflectHelper.GetMethodDefinition<string>(s => s.TrimStart('a')),
+				ReflectHelper.GetMethodDefinition<string>(s => s.TrimStart('a', 'a')),
+				ReflectHelper.GetMethodDefinition<string>(s => s.TrimEnd()),
+				ReflectHelper.GetMethodDefinition<string>(s => s.TrimEnd('a')),
+				ReflectHelper.GetMethodDefinition<string>(s => s.TrimEnd('a', 'a'))
```

Comment:
It looks like this file changes could be useful even without a .NetStandard/.NetCore version, or am I wrong? They may deserve a dedicated PR.

Reply:
No. These are new methods added in .NET Core 2.0 (the optimization which accepts one character), but they are not in full .NET. These changes, however will work in full .NET.